src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/components/SampleFiltering/components/CollectionDateFilter/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import { getDateRangeLabel } from "src/common/utils/dateUtils";
+import { MENU_OPTION_ALL_TIME } from "src/components/DateFilterMenu/constants";
+import { StyledDateFilterMenu, StyledInputDropdown } from "./style";
+
+interface Props {
+  fieldKeyEnd: string;
+  fieldKeyStart: string;
+  menuOptions: DateMenuOption[];
+  updateDateFilter: UpdateDateFilterType;
```

Comment:
what is the `updateDateFilter` prop doing exactly? It's a little hard to infer from the name/usage

Reply:
In the original implementation, the `updateDateFilter` prop was used to register updates to the state of a single filter (eg, collection date range) with the filter state manager. For example, the filter panel for sample table has one logical filtering manager that figures out how to combine all the active filters into a set of filtering rules that yield only those samples that match all the required restrictions. 

However, this case is simpler and doesn't require a filter manager. I will rename this to something more generic when  I do the next PR.

Reply:
ah ok got it, thanks for explaining